test/test_tls.rb

```diff
@@ -0,0 +1,63 @@
+require_relative 'helper'
+require 'fluent/tls'
+
+class UniqueIdTest < Test::Unit::TestCase
+  TEST_TLS1_1_CASES = {
+    'New TLS v1.1' => :'TLS1_1',
+    'Old TLS v1.1' => :'TLSv1_1',
+  }
+  TEST_TLS1_2_CASES = {
+    'New TLS v1.2' => :'TLS1_2',
+    'Old TLS v1.2' => :'TLSv1_2'
+  } 
+  TEST_TLS_CASES = TEST_TLS1_1_CASES.merge(TEST_TLS1_2_CASES)
+
+  sub_test_case 'constants' do
+    test 'default version' do
+      assert_equal :'TLSv1_2', Fluent::TLS::DEFAULT_VERSION
+    end
+
+    data(TEST_TLS_CASES)
+    test 'supported versions' do |ver|
+      assert_include Fluent::TLS::SUPPORTED_VERSIONS, ver
+    end
+
+    test 'default ciphers' do
+      assert_equal "ALL:!aNULL:!eNULL:!SSLv2", Fluent::TLS::CIPHERS_DEFAULT
+    end
+  end
+
+  sub_test_case 'set_version_to_context' do
+    setup do
+      @ctx = OpenSSL::SSL::SSLContext.new
+    end
+
+    data(TEST_TLS_CASES)
+    test 'with version' do |ver|
+      assert_nothing_raised {
+        Fluent::TLS.set_version_to_context(@ctx, ver, nil, nil)
+      }
+    end
+
+    data(TEST_TLS_CASES)
+    test 'can specify old/new syntax to min_version/max_version' do |ver|
+      omit "min_version=/max_version= is not supported" unless Fluent::TLS::MIN_MAX_AVAILABLE
+
+      assert_nothing_raised {
+        Fluent::TLS.set_version_to_context(@ctx, Fluent::TLS::DEFAULT_VERSION, ver, ver)
+      }
```

Comment:
I think it needs to assert `@ctx.min_version` and `@ctx.max_version` is set the given value(Does what the method does set min and max version of TSL?).
The test is just checking this test does not raise an error and it doesn't check the behaivor of Fluent::TLS.set_version_to_context.

Reply:
> I think it needs to assert @ctx.min_version and @ctx.max_version is set the given value(Does what the method does set min and max version of TSL?).

I agree. But the problem is `SSLContext` doesn't provide accessor for `min_version`/`max_version`. If openssl gem implements it, we can add it.